### PR TITLE
CaRT729: Pulling in HG_Forward fixes for Mercury

### PR DESCRIPTION
**Pull request by vschhabra**

### CODE REVIEW

File: mercury.spec [hunk 1]
```diff
@@ -118,6 +118,9 @@ cd build
 
 
 %changelog
+* Thu Mar 05 2020 Vikram Chhabra <vikram.chhabra@intel.com> - 2.0.0a1-0.4
+- Udated to latest master with HG_Forward fix.
```
Comment (brianjmurrell) [inline mercury.spec]:
```suggestion
- Updated to latest master with HG_Forward fix.
```

Comment (vschhabra) [reply mercury.spec]: Sorry my bad. Fixed in latest push

Comment (vschhabra) [reply mercury.spec]: Also fixed release number

**Comment by brianjmurrell** [comment]
I notice that mercury in cart@63fa727c055c2446f4f6f2d06d3aec8e84071c2b which is where daos is currently pointing at is still:

```
MERCURY = 9889a0d0ba8854c1a0c2bd186a01fac4f57edada
```

which is actually now 2 revs behind this module.  Cart's RPM is also restricting mercury to:

```
%global mercury_version 1.0.1-21%{?dist}
```

So I wonder how the changes being proposed to the mercury RPM here have been validated that they work with cart and don't introduce any regressions.  Is there a PR for cart with this mercury commit?

Generally updating the RPM should be the second last step in the process of validating a component change and landing it the last stage.  Prior to that, I would think that the git commit that the RPM is being bumped to has been tested with a cart PR.

**Comment by vschhabra** [comment]
https://github.com/daos-stack/cart/pull/365 is the corresponding PR for cart. It cannot pass the RPM builds and go into CI it seems till this commit happens? Correct me if I am wrong.
I did local tests but cannot land the above until RPM builds pass.

**Comment by vschhabra** [comment]
> I notice that mercury in cart@63fa727c055c2446f4f6f2d06d3aec8e84071c2b which is where daos is currently pointing at is still:
> 
> ```
> MERCURY = 9889a0d0ba8854c1a0c2bd186a01fac4f57edada
> ```
> 
> which is actually now 2 revs behind this module. Cart's RPM is also restricting mercury to:
> 
> ```
> %global mercury_version 1.0.1-21%{?dist}
> ```
> 
> So I wonder how the changes being proposed to the mercury RPM here have been validated that they work with cart and don't introduce any regressions. Is there a PR for cart with this mercury commit?
> 
> Generally updating the RPM should be the second last step in the process of validating a component change and landing it the last stage. Prior to that, I would think that the git commit that the RPM is being bumped to has been tested with a cart PR.



> I notice that mercury in cart@63fa727c055c2446f4f6f2d06d3aec8e84071c2b which is where daos is currently pointing at is still:
> 
> ```
> MERCURY = 9889a0d0ba8854c1a0c2bd186a01fac4f57edada
> ```
> 
> which is actually now 2 revs behind this module. Cart's RPM is also restricting mercury to:
> 
> ```
> %global mercury_version 1.0.1-21%{?dist}
> ```
> 
> So I wonder how the changes being proposed to the mercury RPM here have been validated that they work with cart and don't introduce any regressions. Is there a PR for cart with this mercury commit?
> 
> Generally updating the RPM should be the second last step in the process of validating a component change and landing it the last stage. Prior to that, I would think that the git commit that the RPM is being bumped to has been tested with a cart PR.

@brianjmurrell 
daos-stack/cart#365 is the corresponding PR for cart. It cannot pass the RPM builds and go into CI it seems till this commit happens? Correct me if I am wrong.
I did local tests but cannot land the above until RPM builds pass.

**Comment by brianjmurrell** [comment]
Yes, so what you can do is add a **temporary** reference to this mercury RPM PR to your cart PR by adding:

```
PR_REPOS := mercury@PR-39
```
after line 5 of 
https://github.com/daos-stack/cart/blob/8a12cd814f2ba4834c7fede180ff0ae1a5090cf8/utils/rpms/Makefile#L5

But **NOTE: do NOT land the above change**.  It is only for getting the cart PR to complete RPM build and test.  Once it tests successfully, we can land this PR and then remove the above reference and then land the cart PR.

You are also going to want to update:
https://github.com/daos-stack/cart/blob/8a12cd814f2ba4834c7fede180ff0ae1a5090cf8/utils/rpms/cart.spec#L3 in your cart PR to use the version of mercury that this PR is creating.

**Comment by vschhabra** [comment]
All tests have passed on CI.
https://github.com/daos-stack/cart/pull/365#issuecomment-595924782
Do we merge this one and then I initiate a new build/test on my PR on cart?

**Comment by brianjmurrell** [comment]
> All tests have passed on CI.

Excellent

> Do we merge this one and then I initiate a new build/test on my PR on cart?

Yes, I will go ahead and merge it now.

**Pull request merged**